Library/Homebrew/cask/artifact/symlinked.rb

```diff
@@ -64,19 +64,20 @@ def link(force: false, **options)
         end
 
         ohai "Linking #{self.class.english_name} '#{source.basename}' to '#{target}'"
-        create_filesystem_link(**options)
+        create_filesystem_link(command: command)
       end
 
       def unlink(**)
         return unless target.symlink?
 
         ohai "Unlinking #{self.class.english_name} '#{target}'"
-        target.delete
+        Utils.gain_permissions_remove(target, command: command)
```

Comment:
Looking inside `gain_permissions_remove`, I'm not convinced this behaves correctly - shouldn't the symlink condition come first?

Reply:
Yeah, I think we want to handle `file?` and `symlink?` separately in there.